### PR TITLE
kvserver: log most lease applications

### PR DESCRIPTION
**Pull request by arulajmani**
We now log every lease application, instead of hiding this behind a verbosity setting. To avoid the logs from getting too spammy, we do so every 5 seconds.

Informs #134640

Release note: None

### CODE REVIEW

**Comment by arulajmani** [comment]
@nvanbenschoten I think this should be good for a look. I smoke tested this on a 3 node cluster and the logs weren't too spammy.

File: pkg/kv/kvserver/replica_proposal.go [hunk 1]
```diff
@@ -387,12 +389,31 @@ func (r *Replica) leasePostApplyLocked(
...
+		if r.store.IsDraining() && leaseChangingHands {
+			// If the new leaseholder is on a draining node, in which case it should be
+			// shedding leases, indicate this in the log line.
+			log.Infof(ctx, "new range lease %s on draining node following %s", newLease, prevLease)
+		} else if log.V(1) {
+			// Log every lease acquisition if verbose logging is enabled.
+			log.Infof(ctx, "new range lease %s following %s", newLease, prevLease)
+		} else if ((newLease.Type() != roachpb.LeaseExpiration) ||
+			(newLease.Type() == roachpb.LeaseExpiration && leaseChangingHands)) &&
+			leaseAcquisitionLoggerEvery.ShouldLog() {
```
[the remark this answers is not in the document]
Comment (nvanbenschoten) [reply pkg/kv/kvserver/replica_proposal.go:401]: :100: this is exactly what I was thinking. It's pretty bad when some instances of a log trick a debugger into thinking that the log should always be there and therefore for its absence to be a positive indication that the event hasn't happened more, only to find that random events were being filtered out.

Maybe we can add a `log.EveryN.ShouldLogWithDroppedCount` method, or something along those lines, so as to not break all existing callers at once.

**Review by nvb — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135115)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @stevendanna)

___
*[`pkg/kv/kvserver/replica_proposal.go` line 315 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfAuSq4V91ywtKBLap:-OBfAuSq4V91ywtKBLaq:b-ch2puu) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L315)):*
> ```Go
> )
> 
> var leaseAcquisitionLoggerEvery = log.Every(5 * time.Second)
> ```

Consider dropping this to every 1 second. That should still be reasonably muted.
___
*[`pkg/kv/kvserver/replica_proposal.go` line 391 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfCQ1o10VStps1tAYI:-OBfCQ1o10VStps1tAYJ:b-15lf1v) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L391)):*
> ```Go
> 	leaseChangingHands := prevLease.Replica.StoreID != newLease.Replica.StoreID || prevLease.Sequence != newLease.Sequence
> 
> 	if iAmTheLeaseHolder {
> ```

Do we want to combine this logic with `maybeLogLeaseAcquisition` at some some? Could just be a TODO for now.
___
*[`pkg/kv/kvserver/replica_proposal.go` line 400 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfB9cP2d7U84ZA-_AU:-OBfB9cP2d7U84ZA-_AV:b-yjc1vv) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L400)):*
> ```Go
> 			log.Infof(ctx, "new range lease %s following %s", newLease, prevLease)
> 		} else if ((newLease.Type() != roachpb.LeaseExpiration) ||
> 			(newLease.Type() == roachpb.LeaseExpiration && leaseChangingHands)) &&
> ```

nit: the condition would be easier to understand if you pulled out an `extension := newLease.Type() == roachpb.LeaseExpiration && !leaseChangingHand` variable above, and then wrote this as `} else if !extension &&  leaseAcquisitionLoggerEvery.ShouldLog() {`.
___
*[`pkg/kv/kvserver/replica_proposal.go` line 409 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfBcnL-QjEH9Iv9pRK:-OBfBcnL-QjEH9Iv9pRL:b3u6j7c) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L409)):*
> ```Go
> 			// TODO(arul): consider pulling this out into its own logging channel.
> 			log.Infof(ctx, "new range lease %s following %s", newLease, prevLease)
> 		} else if prevLease.Type() == roachpb.LeaseExpiration && newLease.Type() != roachpb.LeaseExpiration && !leaseChangingHands {
> ```

Same thing about extracting a variable above to make this logic easier to understand.

File: pkg/kv/kvserver/replica_proposal.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (arulajmani) [reply pkg/kv/kvserver/replica_proposal.go]: I'll pull this out in a separate PR.

**Review by arulajmani — COMMENTED** [review]
RFAL

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-:-OBfo51mAF7btwcbKvgD:bb72pd0)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten and @stevendanna)

___
*[`pkg/kv/kvserver/replica_proposal.go` line 315 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfAuSq4V91ywtKBLap:-OBfnqxL0lyTSGii0w6Y:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L315)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Consider dropping this to every 1 second. That should still be reasonably muted.
</blockquote></details>

Done.
___
*[`pkg/kv/kvserver/replica_proposal.go` line 391 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfCQ1o10VStps1tAYI:-OBflYqq20MWsq3W-4Rv:bs7wumx) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L391)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Do we want to combine this logic with `maybeLogLeaseAcquisition` at some some? Could just be a TODO for now.
</blockquote></details>

Done. 
___
*[`pkg/kv/kvserver/replica_proposal.go` line 400 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfB9cP2d7U84ZA-_AU:-OBfm-D95J7drNmEppwj:b-oz8tym) ([raw file](https://github.com/cockroachdb/cockroach/blob/b78f718adaccf8d1b708ad9a1cbe2489fcc19b66/pkg/kv/kvserver/replica_proposal.go#L400)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

nit: the condition would be easier to understand if you pulled out an `extension := newLease.Type() == roachpb.LeaseExpiration && !leaseChangingHand` variable above, and then wrote this as `} else if !extension &&  leaseAcquisitionLoggerEvery.ShouldLog() {`.
</blockquote></details>

Good call.

**Review by nvb — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-:-OBfsDGb-DUj-SKECMwK:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @arulajmani and @stevendanna)

___
*[`pkg/kv/kvserver/replica_proposal.go` line 609 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfs7dc3g2Nh4JVj6qq:-OBfs7ddB3WoPJss3yzW:b7v5os6) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cc2a9d6020f0ac37b22d9a7cd26217ba260310b/pkg/kv/kvserver/replica_proposal.go#L609)):*
> ```Go
> 		// If the new leaseholder is on a draining node, in which case it should be
> 		// shedding leases, indicate this in the log line.
> 		log.Infof(ctx, "new range lease %s on draining node following %s", newLease, prevLease)
> ```

Should these go to `log.Health.Infof` like they do below?
___
*[`pkg/kv/kvserver/replica_proposal.go` line 614 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfrpxb3vlmq36kPPZr:-OBfrpxb3vlmq36kPPZs:bi1sbmi) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cc2a9d6020f0ac37b22d9a7cd26217ba260310b/pkg/kv/kvserver/replica_proposal.go#L614)):*
> ```Go
> 		log.Infof(ctx, "new range lease %s following %s", newLease, prevLease)
> 	} else if !extension && leaseAcquisitionLoggerEvery.ShouldLog() {
> 		// We log lease applications once every 5 seconds
> ```

"every 1 second"

Or just find a way to reword this to eliminate the reference to the specific duration.

**Review by arulajmani — COMMENTED** [review]
TFTR!

bors r=nvanbenschoten

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-:-OBfwheF4IlMIDOTf3el:b-l0r5kv)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten and @stevendanna)

___
*[`pkg/kv/kvserver/replica_proposal.go` line 609 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfs7dc3g2Nh4JVj6qq:-OBfwULvBWJ5wioi-EwK:bzb7lzt) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cc2a9d6020f0ac37b22d9a7cd26217ba260310b/pkg/kv/kvserver/replica_proposal.go#L609)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Should these go to `log.Health.Infof` like they do below?
</blockquote></details>

Done. I assume you mean just for the draining case, right?
___
*[`pkg/kv/kvserver/replica_proposal.go` line 614 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfrpxb3vlmq36kPPZr:-OBfwY5iE5_1zrCF8wIP:btsjbub) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cc2a9d6020f0ac37b22d9a7cd26217ba260310b/pkg/kv/kvserver/replica_proposal.go#L614)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

"every 1 second"

Or just find a way to reword this to eliminate the reference to the specific duration.
</blockquote></details>

Reworded.

**Review by nvb — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135115)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani and @stevendanna)

___
*[`pkg/kv/kvserver/replica_proposal.go` line 609 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfs7dc3g2Nh4JVj6qq:-OBfx5oGAQwO_q-nHY2a:b-6zzi1y) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cc2a9d6020f0ac37b22d9a7cd26217ba260310b/pkg/kv/kvserver/replica_proposal.go#L609)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Done. I assume you mean just for the draining case, right?
</blockquote></details>

Oh no, I meant every case.

**Review by arulajmani — COMMENTED** [review]
bors r=nvanbenschoten

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-:-OBfx_vAC9sQQrRZkeQ4:b-nvg7rs)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten and @stevendanna)

___
*[`pkg/kv/kvserver/replica_proposal.go` line 609 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/135115#-OBfs7dc3g2Nh4JVj6qq:-OBfxZ-X8rtgfIXXMghw:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cc2a9d6020f0ac37b22d9a7cd26217ba260310b/pkg/kv/kvserver/replica_proposal.go#L609)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Oh no, I meant every case.
</blockquote></details>

Fixed.
